crates/bevy_ecs/src/storage/blob_vec.rs

```diff
@@ -223,6 +223,10 @@ impl BlobVec {
         }
         self.len = new_len;
         // Cannot use get_unchecked here as this is technically out of bounds after changing len.
+        // SAFETY:
+        // - `new_len` is less than the old len, so it must fit in this vector's allocation.
```

Comment:
This does not account for the case of underflow. Actually we don't mention that this shouldn't be called on an empty BlobVec in the safety invariants at all.

Reply:
> It is the caller's responsibility to ensure that `index` is less than `self.len()`.

if `self.len()` is 0 then it is not possible to pass in an `index` that is less than `self.len()`

Reply:
Ah OK fair. Still think it's worth mentioning something along these lines in this comment though.

Reply:
I added a comment about this to the declaration of `new_len`.